File: backend/app/schemas/health.py
from pydantic import BaseModel


class ComponentHealth(BaseModel):
    name: str
    status: str  # "ok" | "error" | "not_configured"
    detail: str | None = None


class HealthResponse(BaseModel):
    status: str  # "ok" | "degraded"
    app_name: str
    env: str
    components: list[ComponentHealth]

File: frontend/types/health.ts
// Mirrors backend/app/schemas/health.py — keep in sync as fields change.
export interface ComponentHealth {
  name: string;
  status: "ok" | "error" | "not_configured";
  detail?: string | null;
}

export interface HealthResponse {
  status: "ok" | "degraded";
  app_name: string;
  env: string;
  components: ComponentHealth[];
}
